sdk/platform/http_client/curl/src/az_curl.c

```diff
@@ -519,18 +546,20 @@ static AZ_NODISCARD az_result _az_http_client_curl_send_request_impl_process(
   {
     result = _az_http_client_curl_send_get_request(p_curl);
   }
-  else if (az_span_is_content_equal(p_request->_internal.method, az_http_method_post()))
-  {
-    result = _az_http_client_curl_send_post_request(p_curl, p_request);
-  }
   else if (az_span_is_content_equal(p_request->_internal.method, az_http_method_delete()))
   {
     result = _az_http_client_curl_send_delete_request(p_curl, p_request);
   }
+  else if (az_span_is_content_equal(p_request->_internal.method, az_http_method_post()))
+  {
+    AZ_RETURN_IF_FAILED(_az_http_client_curl_add_expect_header(p_curl, &p_list));
```

Comment:
Why is this not on every request?  Curl is unclear on where they will add "Expect: 100-continue" as a header

Reply:
it will be only always for PUT or for any POST with payload greater than 1024